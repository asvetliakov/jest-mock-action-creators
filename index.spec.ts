import { createDispatchMockImplementation, mockActionCreators } from "./";

it("Creates action creator invokation for given functions or function map", () => {
    const ac1 = jest.fn();
    Object.defineProperty(ac1, "name", { value: "ac1" });
    const ac2 = jest.fn();
    Object.defineProperty(ac2, "name", { value: "ac2" });
    const actions = {
        a: jest.fn(),
        b: jest.fn()
    };

    mockActionCreators(ac1, ac2, actions);

    const res = ac1("arg1", true);
    const res2 = ac2("arg2", false);
    const res3 = actions.a({ a: "hello" });
    const res4 = actions.b({ a: "world", world: true });

    expect(res).toMatchSnapshot();
    expect(res2).toMatchSnapshot();
    expect(res3).toMatchSnapshot();
    expect(res4).toMatchSnapshot();
});

it("Creates dispatch implementation for mocked creators", () => {
    const ac1 = jest.fn();
    Object.defineProperty(ac1, "name", { value: "ac1" });
    const ac2 = jest.fn();
    Object.defineProperty(ac2, "name", { value: "ac2" });
    const actions = {
        a: jest.fn(),
        b: jest.fn()
    };

    mockActionCreators(ac1, ac2, actions);

    const dispatch = jest.fn();

    createDispatchMockImplementation(dispatch, {
        [ac1.name]: 10,
        [ac2.name]: "value",
        a: { a: true },
        b: true
    });

    const res1 = dispatch(ac1());
    const res2 = dispatch(ac2());
    const res3 = dispatch(actions.a());
    const res4 = dispatch(actions.b());

    expect(res1).toBe(10);
    expect(res2).toBe("value");
    expect(res3).toEqual({ a: true });
    expect(res4).toBe(true);
});