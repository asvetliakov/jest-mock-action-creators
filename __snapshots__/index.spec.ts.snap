// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates action creator invokation for given functions or function map 1`] = `
Object {
  "actionCreator": "ac1",
  "args": Array [
    "arg1",
    true,
  ],
}
`;

exports[`Creates action creator invokation for given functions or function map 2`] = `
Object {
  "actionCreator": "ac2",
  "args": Array [
    "arg2",
    false,
  ],
}
`;

exports[`Creates action creator invokation for given functions or function map 3`] = `
Object {
  "actionCreator": "a",
  "args": Array [
    Object {
      "a": "hello",
    },
  ],
}
`;

exports[`Creates action creator invokation for given functions or function map 4`] = `
Object {
  "actionCreator": "b",
  "args": Array [
    Object {
      "a": "world",
      "world": true,
    },
  ],
}
`;
